### PR TITLE
Create warp10 middlewares

### PR DESCRIPTION
**Pull request by FlorentinDUBOIS**

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -51,8 +51,12 @@ if (cluster.isMaster) {
...
       environement variable is not set`);
     process.exit(1);
-  } else
-    logger.info(`[bootstrap] App pepper loaded, ready for cook`);
+  }
+
+  if (!process.env.PROXY_WARP10_URI || !process.env.PROXY_WARP10_WRITE_TOKEN) {
+    logger.error('All warp10 environement variables are not set');
+    process.exit(1);
+  }
```
Comment (miton18) [inline src/main.js:58]: Monitor the app is an option no need to exit :-)

File: src/middlewares/warp10.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    const uri = process.env.PROXY_WARP10_URI;
+    const token = process.env.PROXY_WARP10_WRITE_TOKEN;
+    const labels = [
+      {key: 'method', value: method},
+      {key: 'url', value: url},
+      {key: 'status', value: statusCode}
+    ];
+
+    fetch(`http://freegeoip.net/json/${host}`).then((data) => {
+      const {latitude, longitude} = data;
```
Comment (miton18) [inline src/middlewares/warp10.js:36]: Is it possible to use HTTPS ?

File: src/middlewares/warp10.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+      return fetch(`${uri}/api/v0/update`, {
+        method: 'post',
+        headers: {
+          'Content-Type': 'application/x-www-form-urlencoded',
+          'X-Warp10-Token': token
+        },
+
+        body
+      });
```
Comment (miton18) [inline src/middlewares/warp10.js:48]: Why blank line ?

File: src/api.js [hunk 4]
```diff
@@ -39,6 +40,7 @@ class Api {
         .use(body.json())
         .use(body.urlencoded({extended: true}))
         .use(trafficLogger)
+        .use(warp10)
```
Comment (miton18) [inline src/api.js:43]: Why do you want to monitor mangement API ?

**Review by miton18 — CHANGES_REQUESTED** [review]

File: src/api.js (hunk 4 above)
Comment (FlorentinDUBOIS) [reply src/api.js:43]: Because we have a traffic logger on it

File: src/middlewares/warp10.js (hunk 3 above)
Comment (FlorentinDUBOIS) [reply src/middlewares/warp10.js:48]: I love it !

**Pull request merged**